### PR TITLE
plugins/todo: properly export types and avoid usage of deprecated functions

### PR DESCRIPTION
**Pull request by Rugvip**
Bit of cleanup (:

Looking into what it would be like to fix all warnings in the API reports.

### CODE REVIEW

File: plugins/todo/api-report.md [hunk 1]
```diff
@@ -7,22 +7,84 @@
...
+    value: string;
+  }[];
+};
+
+// @public
+export type TodoListResult = {
+  items: TodoItem[];
+  totalCount: number;
+  offset: number;
+  limit: number;
```
Comment (freben) [inline plugins/todo/api-report.md:84]: wrap these three in a `pageInfo: {}` object, i'd say (and let's find some kind of common ground for how to do these best)

File: plugins/todo/api-report.md [hunk 2]
```diff
@@ -7,22 +7,84 @@
...
+  | 'author'
+  | 'viewUrl'
+  | 'repoFilePath';
+
+// @public
+export type TodoListOptions = {
+  entity?: Entity;
+  offset?: number;
+  limit?: number;
+  orderBy?: {
```
Comment (freben) [inline plugins/todo/api-report.md:69]: i mean, this could be an array ... but that's överkurs i guess

**Review by freben — APPROVED** [review]

File: plugins/todo/api-report.md (hunk 1 above)
Comment (Rugvip) [reply plugins/todo/api-report.md:84]: If we settle on a format at some point, sure. For now I'm not looking to make any changes to the API

File: plugins/todo/api-report.md (hunk 2 above)
Comment (Rugvip) [reply plugins/todo/api-report.md:69]: Same as ^

**Pull request merged**